odoo/tools/pdf.py

```diff
@@ -94,7 +96,14 @@ def add_banner(pdf_stream, text=None, logo=False, thickness=2 * cm):
     :return (BytesIO):              The modified PDF stream.
     """
 
-    old_pdf = PdfFileReader(pdf_stream, strict=False, overwriteWarnings=False)
+    try:
+        old_pdf = PdfFileReader(pdf_stream, strict=False, overwriteWarnings=False)
+        old_pdf.getNumPages()
+    except Exception:
+        # Import is done here to prevent circular import issues.
+        from odoo.tools.translate import _
+        raise UserError(_("Error when reading the original PDF for: %s.\nPlease make sure the file is valid.", text))
```

Comment:
can't `text` be very long? maybe cap it to something like
```suggestion
        raise UserError(_("Error when reading the original PDF for: %r.\nPlease make sure the file is valid.", text[:100]))
```

Reply:
Thanks for the feedback @mart-e 
Good idea :+1: 
It can't hurt to put a cap, indeed :ok_hand:

Reply:
FWIW an even better method is to use the standard library's [`textwrap.shorten`](https://docs.python.org/3/library/textwrap.html?highlight=textwrap#textwrap.shorten), that way you can indicate that the text was cut off (with an ellipsis or somesuch) rather than hard-cut the text (which can be a bit confusing).

Reply:
Never heard of `textwrap`, but thanks a lot for the heads up, it may come in handy at times (like now) :slightly_smiling_face: